unit-test-cloud-functions/functions/test/functions.spec.js

```diff
@@ -0,0 +1,78 @@
+const { expect } = require("chai");
+const admin = require("firebase-admin");
+
+// Initialize the firebase-functions-test SDK using environment variables.
+// These variables are automatically set by firebase emulators:exec
+//
+// This configuration will be used to initialie the Firebase Admin SDK, so
+// when we use the Admin SDK in the tests below we can be confident it will
+// communicate with the emulators, not production.
+const test = require("firebase-functions-test")({
+  projectId: process.env.GCLOUD_PROJECT,
+  databaseURL: process.env.FIREBASE_DATABASE_EMULATOR_HOST,
```

Comment:
Do we really want this line? This project uses only Firestore.

Reply:
Also this is wrong since `FIREBASE_DATABASE_EMULATOR_HOST` is a host (`localhost:5000`) not a URL.

Reply:
Ah good catch, thank you!